code/modules/mob/living/carbon/human/species/_species.dm

```diff
@@ -221,22 +221,24 @@
 
 /datum/species/proc/movement_delay(mob/living/carbon/human/H)
 	. = 0	//We start at 0.
-	var/flight = 0	//Check for flight and flying items
-	var/ignoreslow = 0
-	var/gravity = 0
 
-	if(H.flying)
-		flight = 1
+	if(has_gravity(H))
+		if(H.status_flags & GOTTAGOFAST)
+			. -= 1
 
-	if((H.status_flags & IGNORESLOWDOWN) || (RUN in H.mutations))
-		ignoreslow = 1
+		var/ignoreslow = FALSE
+		if((H.status_flags & IGNORESLOWDOWN) || (RUN in H.mutations))
+			ignoreslow = TRUE
 
-	if(has_gravity(H))
-		gravity = 1
+		if(ignoreslow)
+			if(speed_mod < 0)	// Apply the speed_mod if it's negative. Meaning it's a speedup
+				. += speed_mod
+			return .
```

Comment:
This section can be made a bit more efficient by removing the use of `ignoreslow`:
```dm
		if((H.status_flags & IGNORESLOWDOWN) || (RUN in H.mutations))
			if(speed_mod < 0)	// Apply the speed_mod if it's negative. Meaning it's a speedup
				. += speed_mod
			return .
```

Reply:
Yeah that will remove the var. I did think of other cases though where worn equipment speeds you up. So I'll need to factor those in as well. Which most likely means I'll still be using this var

Reply:
Fair enough